### ISSUE TITLE
[Studio] Site creating failed with git remote when repo has case sensitive files

### ISSUE THREAD

**Issue by phuongnq** [craftercms/craftercms#4747]
## Describe the bug
In some Linux environments, we can have git files with case sensitive. So loader.js and Loader.js are different files.
While creating new site from git repo with such files, it will throw error for duplicating key while inserting to item_state table. 

## To Reproduce
1. Create a site repo blueprint with case sensitive files. For example: loader.js and Loader.js at the same path in different points of the history v1 (/static-assets/js/loader.js) and v2 (/static-assets/js/Loader.js)
2. Create a new site from Studio with blueprint from step 1
3. Site creation is failed with error (Refer to Logs attachment)

## Expected behavior
New site should be created without error

## Screenshots
![git_case_sensitive_01](https://user-images.githubusercontent.com/2996543/120606210-e666ea00-c489-11eb-8ae3-11c92a7361d1.png)
![git_case_sensitive_02](https://user-images.githubusercontent.com/2996543/120606217-e830ad80-c489-11eb-9a58-4993d8f3aa54.png)

## Logs
[site_creation_error.txt](https://github.com/craftercms/craftercms/files/6589795/site_creation_error.txt)

### Version
3.1.14

### OS
Ubuntu 20.04.2 LTS

### Browser
Chrome

**Comment by russdanner** [comment]
2 files in the same path at the same time with the same name but different cases is _NOT_ supported.

Please re-validate if the same behavior can be reproduced when the case change is across time in the git history.

**Comment by phuongnq** [comment]
@russdanner I can reproduce this with case change in history, please use this repo:
https://github.com/phuongnq/craftercms-casesensitive-history-test

Site ID: test
Description: (No description supplied)
Git Repo URL: https://github.com/phuongnq/craftercms-casesensitive-history-test.git
Git Remote Name: origin
Git Branch: master
Sandbox Branch: master

In the repo, I have first commit for both files

```
config/studio/plugins/apps/authoring/src/components/Loader.js
config/studio/plugins/apps/authoring/src/components/loader.js
```

The later commits, I removed both files, then add back `loader.js`.

The error is:
```
Error executing: INSERT INTO item_state (object_id, site, path, state, system_processing) VALUES ('f0c5f2bc-0788-4224-84dc-bd26e5eb0371', 'mis-05', '/config/studio/plugins/apps/authoring/src/components/loader.js', 'NEW_UNPUBLISHED_UNLOCKED', 0)
.  Cause: java.sql.SQLIntegrityConstraintViolationException: (conn:21) Duplicate entry 'mis-05-/config/studio/plugins/apps/authoring/src/components/load' for key 'uq_is_site_path'
Query is : INSERT INTO item_state (object_id, site, path, state, system_processing) VALUES ('f0c5f2bc-0788-4224-84dc-bd26e5eb0371', 'mis-05', '/config/studio/plugins/apps/authoring/src/components/loader.js', 'NEW_UNPUBLISHED_UNLOCKED', 0)

[ERROR] 2021-06-03T21:10:21,792 [http-nio-8080-exec-5] [site.SiteServiceImpl] | Error while creating site: mis-05 ID: mis-05 as clone from remote repository: origin (https://github.com/phuongnq/craftercms-casesensitive-history-test.git). Rolling back. 
org.apache.ibatis.jdbc.RuntimeSqlException: Error executing: INSERT INTO item_state (object_id, site, path, state, system_processing) VALUES ('f0c5f2bc-0788-4224-84dc-bd26e5eb0371', 'mis-05', '/config/studio/plugins/apps/authoring/src/components/loader.js', 'NEW_UNPUBLISHED_UNLOCKED', 0)
.  Cause: java.sql.SQLIntegrityConstraintViolationException: (conn:21) Duplicate entry 'mis-05-/config/studio/plugins/apps/authoring/src/components/load' for key 'uq_is_site_path'
Query is : INSERT INTO item_state (object_id, site, path, state, system_processing) VALUES ('f0c5f2bc-0788-4224-84dc-bd26e5eb0371', 'mis-05', '/config/studio/plugins/apps/authoring/src/components/loader.js', 'NEW_UNPUBLISHED_UNLOCKED', 0)
```

If you create as an orphan (no history), there is no issue.

**Comment by sumerjabri** [comment]
This only happens in 3.1.x

We will change the insert to upsert, as we did in v4, to address this case.

**Comment by dejan-brkic** [comment]
done
PR:
https://github.com/craftercms/studio/pull/2593

**Comment by phuongnq** [comment]
Verified the fix in support/3.1.x
